lib/TPP/Transforms/Utils/VNNIUtils.cpp

```diff
@@ -20,22 +21,54 @@ namespace mlir {
 namespace vnni {
 namespace utils {
 
-std::optional<int64_t> getVnniBlockingFactor(Type type) {
+std::optional<int64_t> getVnniBlockingFactor(Type type, Operation *op) {
   auto elementType = getElementTypeOrSelf(type);
-  if (elementType.isBF16())
+  if (elementType.isBF16()) {
+    // Check if a VNNI factor hint is associated to the IR via DLTI.
+    auto deriveVnniFromDLTI = [&]() -> std::optional<int64_t> {
+      if (!op)
+        return std::nullopt;
+      ModuleOp moduleOp = op->getParentOfType<mlir::ModuleOp>();
+      if (!moduleOp)
+        return std::nullopt;
+      TargetSystemSpecInterface sysSpec = moduleOp.getTargetSystemSpec();
+      if (!sysSpec)
+        return std::nullopt;
+      auto deviceId = StringAttr::get(moduleOp->getContext(), "CPU");
+      auto deviceSpec = sysSpec.getDeviceSpecForDeviceID(deviceId);
+      if (!deviceSpec)
+        return std::nullopt;
+      auto vnniId = StringAttr::get(moduleOp->getContext(), "vnni");
```

Comment:
"VNNI" is an Intel name. The Arm name is "BFDOT" or "BFMLA". This method is making it easier to use the Arm type packing, so we can't expect Arm will use "vnni" in its nomenclature.

Reply:
It is brute forcing our tests to behave predictably.
Our whole infrastructure is VNNI-centric already, no reason to be different here (at least for now).

Reply:
There is: if we have tests for packing factor 4 (Arm), and we use DLTI, we'll have to call it "vnni".

Simply calling it "packing_factor" or "type_packing" would suffice.

Reply:
But then the test pass is still `-pack-vnni` 🙃 
(type_)packing is too generic. I don't mind renaming it per se but vnni is most consistent for now.

Reply:
ok, we can rename vnni to something else in a separate PR